src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManager.cs

```diff
@@ -204,14 +209,61 @@ private static SuppressionDiffResult CompareFiles(SuppressionFile oldFile, Suppr
 		{
 			cancellation.ThrowIfCancellationRequested();
 
-			if (settings.SuppressionMechanismEnabled && Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true)
+			if (settings.SuppressionMechanismEnabled && 
+			    (Instance?.IsSuppressed(semanticModel, diagnostic, cancellation) == true ||
+			    CheckSuppressedComment(diagnostic, cancellation)))
 			{
 				return;
 			}
 
 			reportDiagnostic(diagnostic);
 		}
 
+		private static bool CheckSuppressedComment(Diagnostic diagnostic, CancellationToken cancellation)
+		{
+			cancellation.ThrowIfCancellationRequested();
+			SyntaxNode root = diagnostic.Location.SourceTree.GetRoot();
```

Comment:
cancellation is not passed

Reply:
According to `SourceTree` property description it can be `null` in some cases. So I would use a safe method call:
```suggestion
			SyntaxNode root = diagnostic.Location.SourceTree?.GetRoot();
```